examples/demos/example13-all-features.tsx

```diff
@@ -87,10 +87,10 @@ export default function AllFeaturesExample(): JSX.Element {
       editor: <DropDownEditor<Row> options={titles} column={{} as Column<Row>} onBlur={() => {}} />,
       width: 200,
       resizable: true,
-      cellContentRenderer(props) {
+      formatter(props) {
```

Comment:
Generally speaking, render props is still faster than using a component for basic use case, because if a function only returns a number or string, it avoided unnecessary React related expense. I am wondering if we should do the opposite like removing the `formatter` but keep the `cellContentRenderer`?

In addition to the `formatter`, I believe we should start thinking removing the element support, i.e. at least we should not pass an element as a `formatter` because this could potentially result in not re-render problem.

BTW, not sure if this this working, but I am guessing if we want to put a hook inside this formatter function, it might not work.

Reply:
If you want to use hooks, then you'll just have to separate the component from the formatter function:
```jsx
  formatter(props) {
    return <Formatter {...props} />;
  }
```

Reply:
>Generally speaking, render props is still faster than using a component for basic use case, because if a function only returns a number or string, it avoided unnecessary React related expense. I am wondering if we should do the opposite like removing the formatter but keep the cellContentRenderer?
https://github.com/adazzle/react-data-grid/pull/1855#discussion_r362855946

> In addition to the formatter, I believe we should start thinking removing the element support
I agree we should remove element support. I think we do not 

>at least we should not pass an element as a formatter because this could potentially result in not re-render problem.
I do not think we have to worry about re-rendering formatters anymore. A formatter should only be re-rendered if any of the Cell component props change which should not happen often

>BTW, not sure if this this working, but I am guessing if we want to put a hook inside this formatter function, it might not work.
As formatter is a component so hooks should work

Reply:
@amanmahajan7 
My hook comment above is for if we write code like this

```typescript
function MyGrid() {
const someDataInMyGrid;
const columns = [
  {
    key: xx,
    ...
    formatter(props) {
       const xxx = useState(someDataInMyGrid);
       return someElement;
    }
  }
];

...
}
```

Reply:
This would work because formatter is currently a component. It is same as writing

```
function Formatter() {
   const xxx = useState(someDataInMyGrid);
   return someElement;
}

const columns = [
  {
    key: xx,
    ...
    formatter: Formatter
  }
];
```

This however won't work with render props